runhouse/resources/hardware/utils.py

```diff
@@ -248,3 +258,152 @@ def up_cluster_helper(cluster, capture_output: Union[bool, str] = True):
                     f.write(outfile.output)
     else:
         cluster.up()
+
+
+###################################
+# Cluster list helping methods
+###################################
+
+
+def parse_time_duration(duration: str):
+    # A simple parser for duration like "15m", "2h", "3d"
+    try:
+        unit = duration[-1]
+        value = int(duration[:-1])
+
+        # if the user provides a "--since" time filter than is less than a minute, set it by default to one minute.
+        if unit == "s" and value < 60:
+            value = 60
+
+        time_duration = value * TIME_UNITS.get(unit, 0)
```

Comment:
small nit, but instead of hard coding a constant here can't we use regex, something like? 
`match = re.match(r"(\d+)([smhd])$", value)`